idea/api-gateway/src/rabbitmq/init-rabbitmq.ts

```diff
@@ -0,0 +1,112 @@
+import { connect, Connection, Channel } from 'amqplib';
+import { RabbitMQExchanges, RabbitMQueues } from '@gear-js/common';
+
+import config from '../config/configuration';
+import { RpcResponse } from '../json-rpc/types';
+
+let connectionAMQP: Connection;
+let mainChannelAMQP: Channel;
+
+const testBalanceServicesMap: Map<string, Channel> = new Map<string, Channel>();
+const dataStorageServicesMap: Map<string, Channel> = new Map<string, Channel>();
+const rabbitMQEventMap: Map<string, (params: any) => RpcResponse> = new Map<string, (params: any) => RpcResponse>();
+
+export async function initAMQP(): Promise<void> {
+  try {
+    connectionAMQP = await connect(config.rabbitmq.url);
+    mainChannelAMQP = await connectionAMQP.createChannel();
+
+    await mainChannelAMQP.assertExchange(RabbitMQExchanges.TOPIC_EX, 'topic', { durable: true });
+    await mainChannelAMQP.assertExchange(RabbitMQExchanges.DIRECT_EX, 'direct', { durable: true });
+    const repliesAssertQueue = await mainChannelAMQP.assertQueue(RabbitMQueues.REPLIES, {
+      durable: true,
+      exclusive: false,
+      autoDelete: true,
+      messageTtl: 30_000 });
+
+    await mainChannelAMQP.bindQueue(repliesAssertQueue.queue, RabbitMQExchanges.DIRECT_EX, RabbitMQueues.REPLIES);
+
+    const genesisesAssertQueue = await mainChannelAMQP.assertQueue(RabbitMQueues.GENESISES, {
+      durable: true,
+      exclusive: false,
+      autoDelete: true,
+      messageTtl: 30_000
+    });
+
+    await mainChannelAMQP.bindQueue(genesisesAssertQueue.queue, RabbitMQExchanges.DIRECT_EX, RabbitMQueues.GENESISES);
+    await mainChannelAMQP.bindQueue(genesisesAssertQueue.queue, RabbitMQExchanges.TOPIC_EX, RabbitMQueues.GENESISES);
+
+    await subscribeToGenesises();
+    await subscribeToReplies();
+  } catch (error) {
+    console.error(`${new Date()} | Init rabbitMQ error`, error);
+  }
+}
+
+async function subscribeToReplies(): Promise<void> {
+  await mainChannelAMQP.consume(RabbitMQueues.REPLIES, (message) => {
```

Comment:
Maybe it will be better to create a type for `message` variable?

Reply:
I don't think so. Why need to create message type if `package @types/amqplib` represent this type already.